server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -677,19 +650,37 @@ public void run() {
    * balanceTablets() balances tables by DataLevel. Return the current set of migrations partitioned
    * by DataLevel
    */
-  private static Map<DataLevel,Set<KeyExtent>>
-      partitionMigrations(final Set<KeyExtent> migrations) {
+  private Map<DataLevel,Set<KeyExtent>> partitionMigrations() {
     final Map<DataLevel,Set<KeyExtent>> partitionedMigrations = new EnumMap<>(DataLevel.class);
     // populate to prevent NPE
     for (DataLevel dl : DataLevel.values()) {
-      partitionedMigrations.put(dl, new HashSet<>());
+      Set<KeyExtent> extents = new HashSet<>();
+      // prev row needed for the extent
+      try (var tabletsMetadata = getContext().getAmple().readTablets().forLevel(dl)
+          .fetch(TabletMetadata.ColumnType.PREV_ROW, TabletMetadata.ColumnType.MIGRATION).build()) {
+        for (var tabletMetadata : tabletsMetadata) {
+          if (tabletMetadata.getMigration() != null) {
+            extents.add(tabletMetadata.getExtent());
+          }
+        }
+      }
+      partitionedMigrations.put(dl, extents);
     }
-    migrations.forEach(ke -> {
-      partitionedMigrations.get(DataLevel.of(ke.tableId())).add(ke);
-    });
     return partitionedMigrations;
   }
 
+  /**
+   * Delete the migration, if present, for the given extent if the migration destination is the
+   * provided tserver
+   */
+  void conditionallyDeleteMigration(KeyExtent extent, TServerInstance tserver) {
```

Comment:
May be able to drop this method if the places that called it are removed.

Reply:
Dropped in 0a9455cdea88f167179953a1eb8073bf7a77c2f3